File: src/modules/organization/services/organization-delete-avatar.service.ts
import { prisma } from '../../../infra/database/prisma.service';
import { NotFoundException } from '../../../shared/exceptions';
import { UserRequest } from '../../../shared/types/user-request';
import { userOrganizationValidator } from '../../../shared/validators/user-organization.validator';
import { storedFileDeleteService } from '../../stored-file/services/stored-file-delete.service';
import { OrganizationEntity } from '../domain/entities/organization.entity';

class OrganizationDeleteAvatarService {
  async execute(
    id: number,
    userRequest: UserRequest,
  ): Promise<OrganizationEntity> {
    const organization = await prisma.organization.findFirst({
      where: {
        id: id,
        deletedAt: null,
      },
      include: {
        avatar: true,
      },
    });

    if (!organization)
      throw new NotFoundException('Organização não encontrada!');

    await userOrganizationValidator(id, userRequest);

    const updatedOrganization = await prisma.organization.update({
      data: {
        avatarId: null,
      },
      where: {
        id: organization.id,
      },
      include: {
        avatar: true,
      },
    });

    if (organization.avatar)
      await storedFileDeleteService.execute(organization.avatar.uuid);

    return new OrganizationEntity(updatedOrganization);
  }
}

export const organizationDeleteAvatarService =
  new OrganizationDeleteAvatarService();

File: src/modules/organization/services/organization_delete_avatar_service.py
from src.shared.types.user_request import UserRequest
from src.modules.organization.models import Organization
from src.shared.exceptions.not_found_exception import NotFoundException
from src.shared.validators.user_organization_validator import (
    user_organization_validator,
)
from src.modules.stored_file.services.stored_file_delete_service import (
    StoredFileDeleteService,
)


class OrganizationDeleteAvatarService:
    @classmethod
    def execute(self, id: int, user_request: UserRequest):

        organization = Organization.objects.filter(id=id, deleted_at=None).first()

        if organization is None:
            raise NotFoundException("Organização não encontrada!")

        if organization.avatar is None:
            raise NotFoundException("Arquivo não encontrado!")

        user_organization_validator(id, user_request)

        uuid = organization.avatar.uuid

        organization.avatar = None
        organization.save()

        StoredFileDeleteService.execute(uuid)

        return organization
